codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/HuffmanEncoder.java

```diff
@@ -55,40 +68,83 @@
      *
      * @param out the output stream for the compressed data
      * @param data the string literal to be Huffman encoded
-     * @throws IOException if an I/O error occurs.
-     * @see HuffmanEncoder#encode(OutputStream, byte[], int, int)
      */
-    public void encode(OutputStream out, byte[] data) throws IOException {
-        encode(out, data, 0, data.length);
+    public void encode(ByteBuf out, CharSequence data) {
+        ObjectUtil.checkNotNull(out, "out");
+        if (data instanceof AsciiString) {
+            AsciiString string = (AsciiString) data;
+            try {
+                encodeProcessor.out = out;
+                string.forEachByte(encodeProcessor);
+                encodeProcessor.end();
+
+            } catch (Exception e) {
+                PlatformDependent.throwException(e);
+            }
+        } else {
+            // slow path
```

Comment:
it might make sense to move the "slow path" to its own method, to increase the chances for the fast path to be inlined.